data_utils/raw_dataset_loader.py

```diff
@@ -84,19 +117,33 @@ def __transform_input_to_data_object(self, lines: [str]):
             z_pos = float(node_feat[4].strip())
             node_position_matrix.append([x_pos, y_pos, z_pos])
 
-            num_of_protons = float(node_feat[0].strip())
-            charge_density = float(node_feat[5].strip())
-            magnetic_moment = float(node_feat[6].strip())
-
-            charge_density = charge_density - num_of_protons
-
-            node_feature_matrix.append(
-                [num_of_protons, charge_density, magnetic_moment]
-            )
+            node_feature = []
+            for it in range(len(node_feature_dim)):
+                for icomp in range(node_feature_dim[it]):
+                    it_comp = node_feature_loc_start[it] + icomp
+                    node_feature.append(float(node_feat[it_comp].strip()))
+
+            if LSMS:
+                num_of_protons = node_feature[0]
+                charge_density = node_feature[1]
+                magnetic_moment = float(node_feat[6].strip())
+                charge_density = charge_density - num_of_protons
+                node_feature[1] = charge_density
+                if node_feature_dim[2] == 3:
+                    magnetic_moment_x = float(node_feat[7].strip())
+                    magnetic_moment_y = float(node_feat[8].strip())
+                    magnetic_moment_z = float(node_feat[9].strip())
+                    magnetic_moment_x *= magnetic_moment
```

Comment:
I know that you disagree on adding comments, but explaining why you do 
magnetic_moment_x *= magnetic_moment
would be nice. 
Put yourself in the shoes of a person that is looking at the code for the first time. 
When we talked about this, you said "A person just has to read the code and he/she understands". This code will eventually explore in a lot of capabilities one day. Do you really believe that?

Reply:
You are reviewing an outdated version for this part. It has been updated in the general_improvement branch that has been reviewed and merged by you into the main branch.

Reply:
LOL